mcnp2cad.cpp

```diff
@@ -785,24 +785,53 @@ entity_collection_t GeometryContext::defineCell(  CellCard& cell,  bool defineEm
     case CellCard::SURFNUM:
       {      
         int surface = token.second;
-        bool pos = true;
+        bool positive = true;
         if( surface < 0){
-          pos = false; surface = -surface;
+          positive = false; surface = -surface;
         }
         try{
           SurfaceVolume& surf = makeSurface( deck.lookup_surface_card( surface ) );
-          iBase_EntityHandle surf_handle = surf.define( pos, igm, world_size );
+          iBase_EntityHandle surf_handle = surf.define( positive, igm, world_size );
           stack.push_back(surf_handle);
         }
         catch(std::runtime_error& e) { std::cerr << e.what() << std::endl; }
       }
       break;
     case CellCard::MBODYFACET:
       {
-        int cellnum = std::abs(token.second) / 10;
-        int facet = std::abs(token.second) - (cellnum*10);
-        std::cerr << "Attempting to define facet " << facet << " of cell " << cellnum << std::endl;
-        throw std::runtime_error( "Macrobody facets are not yet supported by mcnp2cad." );
+        int identifier = -std::abs( token.second );
+        int surfacenum = -identifier / 10;
+        int facet = -identifier - ( surfacenum * 10 );
+
+        try{
+          makeSurface( deck.lookup_surface_card( identifier ), NULL, facet);
+          SurfaceVolume& surf = makeSurface( deck.lookup_surface_card( identifier ) );
+          const std::string& mnemonic = deck.lookup_surface_card( identifier )->getMnemonic();
+          bool positive = true;
```

Comment:
We need to make sure we have tests that will exercise all these cases.  For example, an RPP that is centered at the origin  vs an RPP with all points in the positive octant  vs an RPP with all points in the negative octant